### PR TITLE
Fix #218: fix arg names and orders

### LINKED ISSUE #218: Error on demo code execution -- in get_dataset if '.rec' in path: TypeError: argument of type 'NoneType' is not iterable
The following code yields an error. I'm using a Conda environment with Python 3.6 because it is required for the ConfigSpace Python package.

```python
import autogluon as ag
from autogluon import ImageClassification as task

# Load a toy image dataset:
filename = ag.download('http://autogluon-hackathon.s3-website-us-west-2.amazonaws.com/data.zip')
ag.unzip(filename)
dataset = task.Dataset(train_path='data/train')

# Fit classification models:
classifier = task.fit(dataset, epochs=2)

# Evaluate predictions on test data:

test_dataset = task.Dataset('data/test', train=False)
test_acc = classifier.evaluate(test_dataset)
```

This also gives the error:
```python
from autogluon import ImageClassification as task
dataset = task.Dataset(train_path='data/train',
                       test_path='data/test')
classifier = task.fit(dataset,
                      net=ag.space.Categorical('ResNet50_v1b', 'ResNet18_v1b'),
                      time_limits=None,
                      # ngpus_per_trial=1,
                      num_trials = 4)
image = 'data/test/BabyShirt/BabyShirt_323.jpg'
ind, prob = classifier.predict(image)
```

The error:
```
scheduler: FIFOScheduler(
DistributedResourceManager{
(Remote: Remote REMOTE_ID: 0, 
        <Remote: 'inproc://172.18.30.18/28461/1' processes=1 threads=4, memory=7.28 GB>, Resource: NodeResourceManager(4 CPUs, 0 GPUs))
})

Starting Experiments
Num of Finished Tasks is 0
Num of Pending Tasks is 4
 25%|█████████████████████▌                                                                | 1/4 [00:00<00:00,  8.97it/s]Process Process-2:
Traceback (most recent call last):
  File "/home/z/anaconda3/envs/IC2/lib/python3.6/multiprocessing/process.py", line 258, in _bootstrap
    self.run()
  File "/home/z/anaconda3/envs/IC2/lib/python3.6/multiprocessing/process.py", line 93, in run
    self._target(*self._args, **self._kwargs)
  File "/home/z/anaconda3/envs/IC2/lib/python3.6/site-packages/autogluon/scheduler/scheduler.py", line 125, in _worker
    ret = fn(**kwargs)
  File "/home/z/anaconda3/envs/IC2/lib/python3.6/site-packages/autogluon/core/decorator.py", line 52, in __call__
    args = sample_config(args, new_config)
  File "/home/z/anaconda3/envs/IC2/lib/python3.6/site-packages/autogluon/core/decorator.py", line 38, in sample_config
    args_dict[k] = v.init()
  File "/home/z/anaconda3/envs/IC2/lib/python3.6/site-packages/autogluon/core/space.py", line 110, in init
    return self.sample(**config)
  File "/home/z/anaconda3/envs/IC2/lib/python3.6/site-packages/autogluon/core/decorator.py", line 185, in sample
    return self.func(*self.args, **kwargs)
  File "/home/z/anaconda3/envs/IC2/lib/python3.6/site-packages/autogluon/task/image_classification/dataset.py", line 108, in get_dataset
    if '.rec' in path:
TypeError: argument of type 'NoneType' is not iterable

```

### PR DESCRIPTION
**Pull request by zhreshold**
*Issue #, if available:*
#218

*Description of changes:*
FIx potential usage issue similar to #218 

By submitting this pull request, I confirm that you can use, modify, copy, and redistribute this contribution, under the terms of your choice.

### CODE REVIEW

**Comment by szha** [comment]
Job PR-436-1 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-436/1/index.html

File: autogluon/task/image_classification/image_classification.py [hunk 1]
```diff
@@ -27,33 +27,38 @@ class ImageClassification(BaseTask):
     Classifier = Classifier
 
     @staticmethod
-    def Dataset(*args, **kwargs):
-        """Dataset for AutoGluon image classification tasks. 
-           May either be a :class:`autogluon.task.image_classification.ImageFolderDataset`, :class:`autogluon.task.image_classification.RecordDataset`, 
+    def Dataset(path=None, train=True, name=None, input_size=224, crop_ratio=0.875, *args, **kwargs):
```
Comment (jwmueller) [inline autogluon/task/image_classification/image_classification.py:30]: Why is `path` an optional argument (what is role of default value `None`)?

File: autogluon/task/image_classification/image_classification.py [hunk 2]
```diff
@@ -27,33 +27,38 @@ class ImageClassification(BaseTask):
...
+        """Dataset for AutoGluon image classification tasks.
+           May either be a :class:`autogluon.task.image_classification.ImageFolderDataset`, :class:`autogluon.task.image_classification.RecordDataset`,
            or a popular dataset already built into AutoGluon ('mnist', 'fashionmnist', 'cifar10', 'cifar100', 'imagenet').
 
         Parameters
         ----------
+        path : str
```
Comment (jwmueller) [inline autogluon/task/image_classification/image_classification.py:37]: should this say "optional" like it does for the `name` arg?

I think the doc should also state:  "Either `path` or `name` must be specified but not both."
(and I also think these two args should be the first two and train should come later).

**Comment by szha** [comment]
Job PR-436-2 is done. 
Docs are uploaded to http://autogluon-staging.s3-website-us-west-2.amazonaws.com/PR-436/2/index.html

**Pull request merged**